js/fetch_test.ts

```diff
@@ -43,3 +43,142 @@ testPerm({ net: true }, async function fetchBlob() {
+  }
+});
+
+test(function() {
+  try {
+    const h = new Headers(null);
+  } catch (e) {
+    assertEqual(e.message, "Failed to construct 'Headers': Invalid value");
+  }
+});
+
+const headerDict = {
+  "name1": "value1",
+  "name2": "value2",
+  "name3": "value3",
+  "name4": undefined,
+  "Content-Type": "value4"
+};
+const headerSeq = [];
+for (const name in headerDict) {
+  headerSeq.push([name, headerDict[name]]);
+}
+  
+test(function() {
+  const headers = new Headers(headerSeq);
+  for (const name in headerDict) {
+    assertEqual(headers.get(name), String(headerDict[name]));
+  }
+  assertEqual(headers.get("length"), null);
+});
+
+test(function() {
+  const headers = new Headers(headerDict);
+  for (const name in headerDict) {
+    assertEqual(headers.get(name), String(headerDict[name]));
+  }
+});
+
+test(function() {
+  const headers = new Headers(headerDict);
+  const headers2 = new Headers(headers);
+  for (const name in headerDict) {
+    assertEqual(headers2.get(name), String(headerDict[name]));
+  }
+});
+
+test(function() {
+  const headers = new Headers();
+  for (const name in headerDict) {
+    headers.append(name, headerDict[name]);
+    console.log(name, headers.get(name));
+    assertEqual(headers.get(name), String(headerDict[name]));
+  }
+});
+
+test(function() {
+  const headers = new Headers();
+  for (const name in headerDict) {
+    headers.set(name, headerDict[name]);
+    assertEqual(headers.get(name), String(headerDict[name]));
+  }
+});
+
+test(function() {
+  const headers = new Headers(headerDict);
+  for (const name in headerDict) {
+    assert(headers.has(name), "headers has name " + name);
+    /* tslint:disable-next-line:max-line-length */
+    assert(!headers.has("nameNotInHeaders"), "headers do not have header: nameNotInHeaders");
+  }
+  
+});
+
+test(function() {
+  const headers = new Headers(headerDict);
+  for (const name in headerDict) {
+    assert(headers.has(name), "headers have a header: " + name);
+    headers.delete(name);
+    assert(!headers.has(name), "headers do not have anymore a header: " + name);
+  }
+});
+
+test(function() {
+  const headers = new Headers(headerDict);
+  for (const name in headerDict) {
+    assertEqual(headers.get(name), String(headerDict[name]));
+    assertEqual(headers.get("nameNotInHeaders"), null);
+  }
+});
+
+const headerEntriesDict = {
+  "name1": "value1",
+  "Name2": "value2",
+  "name": "value3",
+  "content-Type": "value4",
+  "Content-Typ": "value5",
+  "Content-Types": "value6"
+};
+
+test(function() {
+  const headers = new Headers(headerEntriesDict);
+  const keys = Object.keys(headerEntriesDict);
+  keys.forEach(key => {
+    const value = headerEntriesDict[key];
+    const newkey = key.toLowerCase();
+    headerEntriesDict[newkey] = value;
+  });
+  headers.forEach(function(value, key, container) {
+      assertEqual(headers, container);
+      assertEqual(value, headerEntriesDict[key]);
```

Comment:
You should run ./tools/format.py too. Use arrow notation for anonymous functions.

Reply:
@ry  Done.